docs: add guidelines for localizing angular documentation (#43314)
angular/angular: aio/content/guide/localizing-angular.md

> # Angular Localization Guidelines

Reviewer: This sounds a little ambiguous (it could be confused with guidelines on localizing Angular apps). I t would be good to make it more explicit that this is about localizing the docs.

Reviewer 2: Done.

Angular's localization effort is community-driven.
We receive approximately two to three requests a year from the international Angular community asking for guidance on how to best localize the Angular documentation.

> This document describes Angular's localization guidelines, to help members of the Angular community know what's involved when they translate Angular documentation.

Reviewer: I think it is important to make it clear that people should host the localized version themselves.

Reviewer 2: Added a section, Hosting, that clarifies this.

> 

Reviewer: Since this document is going to be seen by people that are interested in localizing Angular.io content to a language for which there is already an active localization effort, I think it would make sense to also encourage them to check whether there is already a localized version for their target language (for example, point them to the page mentioned in `Awareness`).

Reviewer 2: Done.

> ## Angular Google Developer Expert Sponsorship

Reviewer 2: Changed to "Nomination" to avoid confusion about financial support.

> To be officially recognized as a localized version of Angular documentation, an Angular Google Developer Expert (GDE) must either:

Reviewer: Maybe it would be a good idea to expand a bit on what it means to be "officially recognized". (For example, that we will link to the localized version from the official docs.)

Reviewer 2: Done.

* Be directly responsible for the localization effort
* Sponsor the contributors who are localizing the content

> Sponsorship, in this instance, means that the GDE knows the individual or team who wants to localize the Angular content and can vouch for their professionalism and capabilities. GDEs can indicate their sponsorship by contacting a member of the Angular team by email letting them know of the effort.

Reviewer: As a non-native speaker, I am confused: Does this involve some form of financial support?

Reviewer 2: No, there is no financial support here. I have updated to use the terms nominate/nomination instead.

## What to localize

To localize Angular documentation, you must include, at a minimum, the following topics:

> * [Introduction to the Angular Docs](/docs)

Reviewer: Nit: It is preferable to omit the leading `/` in these links

Reviewer 2: Done.

* [What is Angular?](/guide/what-is-angular)
* [Getting started with Angular](/start)
  * [Adding navigation](/start/start-routing)
  * [Managing data](/start/start-data)
  * [User forms for user input](/start/start-forms)
  * [Deploying an application](/start/start-deployment)

>   * [Setting up the local environment and work  space](/guide/setup-local)

Reviewer: To match the guide's title:

```suggestion
  * [Setting up the local environment and workspace](/guide/setup-local)
```

Reviewer 2: Done.

* [Tour of Heroes app and tutorial](/tutorial)
  * [Create a new project](/tutorial/toh-pt0)
  * [The hero editor](/tutorial//toh-pt1)
  * [Display a selection list](/tutorial/toh-pt2)   
  * [Create a feature component](/tutorial/toh-pt3)
  * [Add services](/tutorial/toh-pt4)
  * [Add navigation with routing](/tutorial/toh-pt5)
  * [Get data from a server](/tutorial/toh-pt6)

Because these topics reflect the minimum documentation set for localization, the Angular writing team will take special precautions when making any changes to these topics. Specifically:

> * The Angular team will carefully assess any incoming pull requests or issues
> to determine their impact on localized content.

Reviewer: Nit:

```suggestion
* The Angular team will carefully assess any incoming pull requests or issues to determine their impact on localized content.
```

Reviewer 2: Done.

* Should the Angular team incorporate changes into these topics, the Angular team will communicate those changes to members of the localization community. See the section, Communications, for more information.

## Awareness

> As part of the localization effort, the Angular writing team will create a new topic, Localized Documentation. This topic will list:

Reviewer 3: ```suggestion
As part of the localization effort, the Angular writing team will create a new topic, "Localized documentation". This topic will list:
```
Can you indicate when this topic indicate is expected to be available?
Is there a better way to indicate that "Localized documentation" is a title?

Reviewer 2: Created the topic and updated accordingly. Thanks!

* The language of the localized documentation
* The URL for the localized documentation
* An email address to contact the individual or team who localized the documentation

The Angular team can remove a link on this page for any reason, including but not limited to:

* Unable to contact the individual or team
* Issues or complaints about the documentation that go unaddressed

## Communication

The Angular documentation team will use a Slack channel to communicate with members of the community focused on localization efforts. After receiving sponsorship to localize content, an individual or team can contact the Angular team to get access to this Slack channel.

The Angular documentation team may also conduct quarterly meetings to discuss localization efforts. For example:

* To discuss additional topics that should be part of the minimum documentation set
* To discuss issues with content language that is difficult to translate/localize

> 

Reviewer: Might be outside the scope of this PR, but at some point we could add a section about setting up a new localization project (for example, re-using some of the `aio/` infrastructure).

It would be a good contribution candidate for one of the people that are already localizing angular.io.

Reviewer 2: Agreed. I think it's out of scope here, but definitely something we should consider adding. How complicated is the process?

Reviewer: No idea :grin:
We should reach out to the maintainers of the existing localized versions to find out :smiley:

# Angular documentation localization guidelines

One way to contribute to Angular's documentation is to localize it into another language. This topic describes what you need to know to localize Angular and have it listed on our [Localized documentation](guide/localized-documentation) page.

## Before you begin

Before you start localizing the Angular documentation, first check to see if a localized version already exists. See [Localized documentation](guide/localized-documentation) for a current list of localized versions of Angular documentation.

## Getting started

To have a localized version of Angular documentation listed on [angular.io](https://angular.io), you must either:

> * Be an Angular Google Developer Expert (GDE)

Reviewer 4: ```suggestion
* Be an Angular [Google Developer Expert (GDE)](https://developers.google.com/community/experts)
```

* Have an Angular GDE nominate you for localizing the content

Nomination, in this instance, means that the GDE knows who you are and can vouch for your capabilities. An Angular GDE can nominate someone by contacting the Angular team, providing your name, contact information, and the language to which you are localizing.

## What to localize

To localize Angular documentation, you must include, at a minimum, the following topics:

* [Introduction to the Angular Docs](docs)
* [What is Angular?](guide/what-is-angular)
* [Getting started with Angular](start)
  * [Adding navigation](start/start-routing)
  * [Managing data](start/start-data)

>   * [User forms for user input](start/start-forms)

Reviewer 3: ```suggestion
  * [Using forms for user input](start/start-forms)
```

  * [Deploying an application](start/start-deployment)
  * [Setting up the local environment and workspace](guide/setup-local)
* [Tour of Heroes app and tutorial](tutorial)
  * [Create a new project](tutorial/toh-pt0)
  * [The hero editor](tutorial/toh-pt1)
  * [Display a selection list](tutorial/toh-pt2)   
  * [Create a feature component](tutorial/toh-pt3)
  * [Add services](tutorial/toh-pt4)
  * [Add navigation with routing](tutorial/toh-pt5)
  * [Get data from a server](tutorial/toh-pt6)

Because these topics reflect the minimum documentation set for localization, the Angular documentation team takes special precautions when making any changes to these topics. Specifically:

* The Angular team carefully assesses any incoming pull requests or issues to determine their impact on localized content.

> 

Reviewer 4: ```suggestion
```

> * Should the Angular team incorporate changes into these topics, the Angular team will communicate those changes to members of the localization community. See the section, [Communications](#communications), for more information.

Reviewer 3: ```suggestion
* If the Angular team incorporates changes into these topics, the Angular team will communicate those changes to members of the localization community. See the section, [Communications](#communications), for more information.
```
"should" could be misleading in this case

## Hosting

> Individuals and teams that localize Angular assume responsibility for hosting their localized site. The Angular team does not host localized content.

Reviewer 3: ```suggestion
Individuals and teams that localize Angular documentation assume responsibility for hosting their localized site. The Angular team does not host localized content.
```

> Individuals and teams that localize Angular assume responsibility for hosting their localized site. The Angular team does not host localized content.

Reviewer 4: Probably also worth mentioning we're not responsible for providing a domain name.

## Awareness

> As part of the localization effort, the Angular documentation team adds localized documentation to [Localized documentation](guide/localized-documentation). This topic lists:

Reviewer 3: ```suggestion
As part of the localization effort, the Angular documentation team adds localized documentation to the [Localized documentation](guide/localized-documentation) page. This topic lists:
```
for clarity

* The language of the localized documentation
* The URL for the localized documentation

The Angular team can remove a link on this page for any reason, including but not limited to:

> * Unable to contact the individual or team

Reviewer 3: ```suggestion
* Inability to contact the individual or team
```
Parallel
